test/e2e/test/test.go

```diff
@@ -58,28 +64,24 @@ type TestCase struct {
 	TestMsg         *TestMSG
 	Client          TestClient
 	Ctx             context.Context
+	NSIncluded      *[]string
 }
 
 var TestClientInstance TestClient
+var isVeleroInstalled bool = false
 
-func TestFunc(test VeleroTest) func() {
+func TestFunc(test VeleroBackupRestoreTest) func() {
 	return func() {
 		var err error
 		TestClientInstance, err = NewTestClient()
 		Expect(err).To(Succeed(), "Failed to instantiate cluster client for backup tests")
 		test.Init()
 		BeforeEach(func() {
 			flag.Parse()
-			if VeleroCfg.InstallVelero {
+			if VeleroCfg.InstallVelero && !isVeleroInstalled {
+				Expect(VeleroUninstall(context.Background(), VeleroCfg.VeleroCLI, VeleroCfg.VeleroNamespace)).To((Succeed()))
 				Expect(VeleroInstall(context.Background(), &VeleroCfg, "", false)).To(Succeed())
-			}
-
-		})
-
-		AfterEach(func() {
-			if VeleroCfg.InstallVelero {
-				err := VeleroUninstall(context.Background(), VeleroCfg.VeleroCLI, VeleroCfg.VeleroNamespace)
-				Expect(err).To(Succeed())
+				isVeleroInstalled = true
 			}
```

Comment:
Here, In the cases which call the TestFunc will only install velero once, but will not uninstall it when complete. Later I will further improve in other modules.